src/api-client.ts

```diff
@@ -20,87 +14,61 @@ const enum ContentTypes {
     SRT = 'application/x-subrip'
 }
 
-export class RevAiApiClient {
+export class RevAiApiClient extends ApiRequestHandler {
```

Comment:
Why did you choose inheritance over composition?

I'm thinking from a testing perspective. To create components that are unit testable, I would prefer composition.
We could have two components: 
1. `RevAiApiClient`
2. `ApiRequestHelper`

The tests for the api methods in RevAiApiClient only need to assert that the methods of ApiRequestHelper were called with the correct parameters.

The tests for ApiRequestHelper assert that axios methods were called with the correct parameters and that different response status codes are handled appropriately.

Reply:
In the present state of your PR RevAiApiClient tests are overextending themselves and testing for implementation logic of the ApiRequestHandler. I think you can avoid this by switching to use composition instead of inheritance and refactoring the tests just a bit.